File: Data Structures & Algorithms/contains-duplicate-ii/submission-1.ts
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {boolean}
     */
    containsNearbyDuplicate(nums: number[], k: number): boolean {
        for(let i = 0; i < nums.length; i++){
            for(let j = i + 1; j < Math.min(nums.length, i + k + 1); j++){
                if(nums[i] === nums[j]){
                    return true;
                }
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/max-consecutive-ones-iii/submission-1.java
class Solution {
    public int longestOnes(int[] nums, int k) {
        int left = 0;
        int max_Length = 0;
        int count_zero = 0;

        for(int right = 0; right < nums.length; right++){
            if(nums[right] == 0){
                count_zero++;
            }
            while(count_zero > k){
                if(nums[left] == 0){
                    count_zero--;
                }
                left++;
            }
           max_Length = Math.max(max_Length, right - left + 1);
        }
        return max_Length;
    }
}